python/mujinplanningclient/handeyecalibrationplanningclient.py

```diff
@@ -2,45 +2,69 @@
+
     tasktype = 'handeyecalibration'
 
+    robot = None
+
+    _deprecated = None # used to mark arguments as deprecated (set argument default value to this)
+
     def __init__(self, robot, **kwargs):
-        """Logs into the mujin controller, initializes hand eye calibration task, and sets up parameters
+        """Connects to the Mujin controller, initializes HandEyeCalibration task and sets up parameters
 
         Args:
-            controllerurl (str): URL of the mujin controller, e.g. http://controller14
+            robot (str): Name of the robot, e.g. VP-5243I
+            taskzmqport (int, optional): Port of the task's ZMQ server, e.g. 7110. (Default: 11000)
+            taskheartbeatport (int, optional): Port of the task's ZMQ server's heartbeat publisher, e.g. 7111. (Default: 11001)
+            taskheartbeattimeout (float, optional): Seconds until reinitializing task's ZMQ server if no heartbeat is received, e.g. 7
+            tasktype (str, optional): Type of the task, e.g. 'binpicking', 'handeyecalibration', 'itlrealtimeplanning3'. Default: handeyecalibration
+            ctx (zmq.Context, optional): Seconds until reinitializing task's ZMQ server if no heartbeat is received, e.g. 7
+            slaverequestid:
+            controllerip (str): IP or hostname of the mujin controller, e.g. 172.17.0.2 or controller123
+            controllerurl (str, optional): (Deprecated. Use controllerip instead) URL of the mujin controller, e.g. http://controller14.
             controllerusername (str): Username for the Mujin controller, e.g. testuser
             controllerpassword (str): Password for the Mujin controller
             scenepk (str, optional): Primary key (pk) of the scene, e.g. irex_demo.mujin.dae
+            callerid (str, optional): Caller identifier to send to server on every command
+            ignoredArgs: Additional keyword args are not used, but allowed for easy initialization from a dictionary
         """
         super(HandEyeCalibrationPlanningClient, self).__init__(tasktype=self.tasktype, **kwargs)
         self.robot = robot
 
+
+    #
+    # Commands
+    #
+
+
     def ComputeCalibrationPoses(self, primarySensorSelectionInfo, secondarySensorSelectionInfos, numsamples, calibboardvisibility, calibboardLinkName=None, calibboardGeomName=None, timeout=3000, gridindex=None, **kwargs):
         """Compute a set of calibration poses that satisfy the angle constraints using latin hypercube sampling (or stratified sampling upon failure)
 
         Args:
             primarySensorSelectionInfo (dict): Selects the primary camera that everything will be calibrated against.
             secondarySensorSelectionInfos (list[dict]): Selects the secondary camera(s) (assumed to be nearby the primary sensor).
             numsamples (int): Number of samples to take. A reasonable number is often between 5 and 25.
-            calibboardvisibility (bool):
+            calibboardvisibility (dict):
```

Comment:
Is this right?

Reply:
Yes. I've double checked it with the server side. It is also dict in another fuction.
```
def ComputeCalibrationPoses(self, **kwargs):
...
        minPoseGridDepth = self.goals['calibboardvisibility']['distances'] ...
```